### ISSUE TITLE
Confused about undefined-variable

### ISSUE THREAD

**Issue by danizen** [PyCQA/pylint#2034]
### Steps to reproduce
1.  Remove your ~/.pylintrc file to assure report is not suppressed
2. Create a file testlinter.py following this public gist - https://gist.github.com/danizen/50a9ca21b462581ca8e23e17f999ee63
3. Run pylint

### Current behavior

`y` is reported as undefined only in the function `real_unitialized_problem`.

### Expected behavior

`y` should be undefined in the functions 'real_uninitialized_problem' and `real_uninitialized_with_if` tests.
In other words, there should be some real code flow analysis.

### pylint --version output

$ pylint --version
Using config file C:\Users\davisda4\.pylintrc
pylint 1.8.4,
astroid 1.6.3
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  6 2015, 01:54:25) [MSC v.1900 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Was there supposed to be an `y` in `real_unitialized_problem`? I can't reproduce the problem you're having unless there is something else I'm missing. Do note though that we don't quite support full control flow analysis, we ignore branches when inferring the potential values, but this behaviour might change in the future.

**Comment by AraHaan** [comment]
![28d8ef94-b47e-4307-82f7-dff84a2c2e44](https://user-images.githubusercontent.com/15173749/39254047-37f49f26-4877-11e8-824b-8dfaf445a0c2.png)
I always seem to be running into this issue with this ci on my git for some reason which is a false positive. And I hate suppressing this stuff that should be no problem anyway. It is like pylint is dumb thinking it is not defined when clearly it is a function and the function name in the \_\_all\_\_ is clearly encased in a string. I do know that in python it does not matter if I single or double quote strings as it is the same thing.

**Comment by AraHaan** [comment]
@PCManticore I get this issue that is similar on Codacy which is a code checking ci I use on my python gits.
